pool/app/logger/logger.ml

```diff
@@ -118,9 +118,99 @@ let format_reporter
   { Logs.report }
 ;;
 
+let get_log_level () =
+  match Sys.getenv_opt "LOG_LEVEL" with
+  | Some "debug" -> Some Logs.Debug
+  | Some "error" -> Some Logs.Error
+  | Some "warning" -> Some Logs.Warning
+  | _ -> Some Logs.Info
+;;
+
+let logs_dir () =
+  match
+    Sihl.Configuration.root_path (), Sihl.Configuration.read_string "LOGS_DIR"
+  with
+  | _, Some logs_dir -> logs_dir
+  | Some root, None -> root ^ "/logs"
+  | None, None -> "logs"
+;;
+
+let lwt_file_reporter ?(pp_header = pp_exec_header) () =
+  let logs_dir = logs_dir () in
+  let buf () =
+    let b = Buffer.create 512 in
+    ( b
+    , fun () ->
+        let m = Buffer.contents b in
+        Buffer.reset b;
+        m )
+  in
+  let app, app_flush = buf () in
+  let err, err_flush = buf () in
+  let report src level ~over k msgf =
+    let k _ = k () in
+    let write () =
+      let name =
+        match[@warning "-4"] level with
+        | Logs.Error -> logs_dir ^ "/error.log"
+        | _ -> logs_dir ^ "/app.log"
```

Comment:
Please be exhaustive here (`| App | Warning | Info | Debug -> ...`) and get rid of suppression of warning 4.

Reply:
Done